contracts/land/LANDRegistry.sol

```diff
@@ -479,4 +472,87 @@ contract LANDRegistry is Storage, Ownable, FullAssetRegistry, ILANDRegistry {
     assembly { size := extcodesize(addr) }
     return size > 0;
   }
+
+  //
+  // LAND Ping
+  //
+
+  /**
+   * @dev Set date from when the LAND Ping feature should be enabled
+   * @param _gracePeriod Desired amount of time in miliseconds to enable feature
+   */
+  function setGracePeriod(uint _gracePeriod) external onlyDeployer {
+    gracePeriod = _gracePeriod;
+  }
+
+  /**
+   * @dev Set amount of time that should pass for a LAND to be transferred to
+   * a new onwer
+   * @param _deemPeriod Desired amount of time in miliseconds for a LAND to decay
+   */
+  function setDeemPeriod(uint _deemPeriod) external onlyDeployer {
+    deemPeriod = _deemPeriod;
+  }
+
+  /**
+   * @dev Ping an address, keep all that address LAND "alive"
+   */
+  function ping(address _user) external {
+    require(
+      _isApprovedForAll(_user, msg.sender) ||
+      updateManager[_user][msg.sender],
+      "This function can only be called by the operatorForAll or updateManager"
+    );
+    // solium-disable-next-line security/no-block-members
+    _ping(_user);
+  }
+
+  /**
+   * @dev Ping myself, keep all my LAND "alive". This is only refresh owned assets, not operated/allowed ones
+   */
+  function pingMyself() external {
+    _ping(msg.sender);
+  }
+
+  function _ping(address _address) private {
+    // solium-disable-next-line security/no-block-members
+    latestPing[_address] = block.timestamp;
+    emit Ping(msg.sender, _address);
+  }
+
+  /**
+   * @dev Check if LAND is still alive or not
+   */
+  function hasDecayed(uint256 _tokenId) external view returns (bool) {
+    return _hasDecayed(_tokenId);
+  }
+
+  function _hasDecayed(uint256 _tokenId) private view returns (bool) {
```

Comment:
I think that this abstraction not worth it, but if we still need it I would suggest using `internal` instead of `private`